src/components/molecules/ExportProgressItem.tsx

```diff
@@ -0,0 +1,615 @@
+import ProgressBar from '../atoms/ProgressBar'
+import styled from '../../lib/styled/styled'
+import { border } from '../../lib/styled/styleFunctions'
+import { usePreferences } from '../../lib/preferences'
+import { usePreviewStyle } from '../../lib/preview'
+import { useToast } from '../../lib/toast'
+import {
+  getPathByName,
+  mkdir,
+  showOpenDialog,
+  writeFile,
+} from '../../lib/electronOnly'
+import {
+  convertNoteDocToHtmlString,
+  convertNoteDocToMarkdownString,
+  convertNoteDocToPdfBuffer,
+  getValidNoteTitle,
+} from '../../lib/exports'
+import { useTranslation } from 'react-i18next'
+import { useDb } from '../../lib/db'
+import { useEffectOnce } from 'react-use'
+import {
+  AllDocsMap,
+  Attachment,
+  FolderDoc,
+  NoteDoc,
+  ObjectMap,
+} from '../../lib/db/types'
+import { entries, keys, values } from '../../lib/db/utils'
+import { join } from 'path'
+import { filenamify } from '../../lib/string'
+
+interface ExportProcedureData {
+  folderName: string
+  folderPathname: string
+  exportType: string
+  recursive: boolean
+  exportingStorage: boolean
+}
+
+interface ExportProgressItemProps {
+  storageId: string
+  exportSettings: ExportProcedureData
+  onFinish: () => void
+  noteCount?: number
+}
+
+interface ExportOperationState {
+  exportOperation: string
+  exportSubOperationTitle: string
+  exportSubOperationContent: string
+  exportProgressValue: number
+}
+
+const ExportProgressItem = ({
+  storageId,
+  exportSettings,
+  onFinish,
+  noteCount = 0,
+}: ExportProgressItemProps) => {
+  const { preferences } = usePreferences()
+  const includeFrontMatter = preferences['markdown.includeFrontMatter']
+  const { previewStyle } = usePreviewStyle()
+  const { pushMessage } = useToast()
+  const { t } = useTranslation()
+  const { storageMap } = useDb()
+
+  const [showExportProgress, setShowExportProgress] = useState(false)
+  const [exportProgressMaxValue] = useState(100)
+  const [exportErrors, setExportErrors] = useState([] as string[])
+  const [exportState, setExportState] = useState({
+    exportOperation: '',
+    exportSubOperationTitle: '',
+    exportSubOperationContent: '',
+    exportProgressValue: 0,
+  })
+  const canceledRef = useRef<boolean>(false)
+
+  const [dialogIsOpen, setDialogIsOpen] = useState(false)
+  const openDialog = useCallback(async () => {
+    if (dialogIsOpen) {
+      return
+    }
+    setDialogIsOpen(true)
+    try {
+      const result = await showOpenDialog({
+        properties: ['openDirectory', 'createDirectory'],
+        buttonLabel: t('folder.select'),
+        defaultPath: getPathByName('home'),
+      })
+      if (result.canceled) {
+        return
+      }
+      if (result.filePaths == null) {
+        return
+      }
+
+      return result.filePaths[0]
+    } catch (error) {
+      throw error
+    } finally {
+      setDialogIsOpen(false)
+    }
+  }, [dialogIsOpen, t])
+
+  const updateExportState = useCallback(
```

Comment:
use useSetState hook of `react-use` https://github.com/streamich/react-use/blob/master/docs/useSetState.md

Reply:
Thx didn't know about it, used!